classes/class-wc-connect-jetpack.php

```diff
@@ -159,40 +115,57 @@ public static function tracks_record_event( $user, $event_type, $data ) {
 		 * @return bool Whether or nor the current user is connected to Jetpack
 		 */
 		public static function is_current_user_connected() {
-			if ( class_exists( '\Automattic\Jetpack\Connection\Manager' ) && method_exists( '\Automattic\Jetpack\Connection\Manager', 'is_user_connected' ) ) {
-				$connection = new Manager();
-
-				return $connection->is_user_connected();
-			}
-
-			if ( defined( 'JETPACK_MASTER_USER' ) ) {
-				$user_token = self::get_master_user_access_token( JETPACK_MASTER_USER );
-
-				return ( isset( $user_token->external_user_id ) && get_current_user_id() === $user_token->external_user_id );
-			}
-
-			return false;
+			return self::get_connection_manager()->is_user_connected();
 		}
 
 		/**
-		 * Determines if Jetpack is connected
+		 * Determines if Jetpack is connected for the plugin
 		 *
 		 * @return bool Whether or nor Jetpack is connected
 		 */
 		public static function is_connected() {
-			if ( class_exists( '\Automattic\Jetpack\Connection\Manager' ) && method_exists( '\Automattic\Jetpack\Connection\Manager', 'is_connected' ) ) {
-				$connection = new Manager();
-
-				return $connection->is_connected();
-			}
+			$manager = self::get_connection_manager();
 
-			if ( defined( 'JETPACK_MASTER_USER' ) ) {
-				$user_token = self::get_master_user_access_token( JETPACK_MASTER_USER );
+			return $manager->is_plugin_enabled() && self::is_active() && ! $manager->is_missing_connection_owner();
```

Comment:
`is_plugin_enabled()` is a rudimentary piece left from the "Soft Disconnect" times, deprecated now.
`is_missing_connection_owner()` is also excessive if you check for `has_connected_owner()` inside `is_active()`, as they both check for the following:
- connection owner is set
- they have a user token
- WP user account actually exists for their userID

So it loooks like `is_connected()` and `is_active()` might be serving the same purpose, and deprecating one of them wouldn't hurt.